crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -474,18 +504,18 @@ impl TryFrom<types::PaymentsCancelResponseRouterData<DataTransCancelResponse>>
         item: types::PaymentsCancelResponseRouterData<DataTransCancelResponse>,
     ) -> Result<Self, Self::Error> {
         let status = match item.response {
-            DataTransCancelResponse::Empty => match item.http_code {
-                204 => common_enums::AttemptStatus::Voided,
-                _ => common_enums::AttemptStatus::Failure,
-            },
+            DataTransCancelResponse::Empty => {
+                if item.http_code == 204 {
+                    common_enums::AttemptStatus::Voided
+                } else {
+                    common_enums::AttemptStatus::Failure
+                }
+            }
             DataTransCancelResponse::Error(error) => {
-                match error
-                    .message
-                    .unwrap_or(consts::NO_ERROR_MESSAGE.to_string())
-                    .as_str()
-                {
-                    "transaction already canceled" => common_enums::AttemptStatus::Voided,
-                    _ => common_enums::AttemptStatus::Failure,
+                if error.message == *"transaction already canceled" {
```

Comment:
```suggestion
                if error.message == "transaction already canceled".to_string() {
```

Reply:
Please maintain constants for such strings